### PR TITLE
Typecheck the tests

### PR DESCRIPTION
**Pull request by kierdavis**
A recurring problem I've noticed is our tests getting out of sync with subtle changes to our API, which reduces my confidence that the tests are checking what we're expecting them to be checking. Hopefully, holding the tests to the same standard of type-safety as our main codebase would help to improve the situation.

I've tried to group similar changes into commits, so I'd recommend reviewing this PR on a commit-by-commit basis.

### CODE REVIEW

File: tests/components/test_servo.py [hunk 1]
```diff
@@ -1,11 +1,14 @@
 """Tests for the servo classes."""
-from typing import List, Optional, Type
+from typing import TYPE_CHECKING, List, Optional, Type
 
 import pytest
 
 from j5.boards import Board
 from j5.components.servo import Servo, ServoInterface, ServoPosition
 
+if TYPE_CHECKING:  # pragma: nocover
```
Comment (trickeydan) [inline tests/components/test_servo.py]: Not sure if we need `# pragma: nocover` is needed here.

File: tests/components/test_power_output.py [hunk 2]
```diff
@@ -1,9 +1,12 @@
 """Tests for the power output classes."""
-from typing import List, Optional, Type
+from typing import TYPE_CHECKING, List, Optional, Type
 
 from j5.boards import Board
 from j5.components.power_output import PowerOutput, PowerOutputInterface
 
+if TYPE_CHECKING:  # pragma: nocover
```
Comment (trickeydan) [inline tests/components/test_power_output.py]: Not sure if we need `# pragma: nocover` is needed here.

File: tests/components/test_piezo.py [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
 from datetime import timedelta
-from typing import List, Optional, Type
+from typing import TYPE_CHECKING, List, Optional, Type
 
 import pytest
 
 from j5.boards import Board
 from j5.components.piezo import Note, Piezo, PiezoInterface
 
+if TYPE_CHECKING:  # pragma: nocover
```
Comment (trickeydan) [inline tests/components/test_piezo.py]: Not sure if we need `# pragma: nocover` is needed here.

File: tests/components/test_led.py [hunk 4]
```diff
@@ -1,9 +1,12 @@
 """Tests for the LED Classes."""
-from typing import List, Optional, Type
+from typing import TYPE_CHECKING, List, Optional, Type
 
 from j5.boards import Board
 from j5.components.led import LED, LEDInterface
 
+if TYPE_CHECKING:  # pragma: nocover
```
Comment (trickeydan) [inline tests/components/test_led.py]: Not sure if we need `# pragma: nocover` is needed here.

File: tests/components/test_battery_sensor.py [hunk 5]
```diff
@@ -1,10 +1,12 @@
 """Tests for the Battery Sensor Classes."""
-from typing import List, Optional, Type
+from typing import TYPE_CHECKING, List, Optional, Type
 
 from j5.boards import Board
-from j5.components import Component
 from j5.components.battery_sensor import BatterySensor, BatterySensorInterface
 
+if TYPE_CHECKING:  # pragma: nocover
```
Comment (trickeydan) [inline tests/components/test_battery_sensor.py]: Not sure if we need `# pragma: nocover` is needed here.

File: tests/boards/test_board.py [hunk 6]
```diff
@@ -1,11 +1,14 @@
 """Test the base classes for boards."""
-from typing import List, Optional
+from typing import TYPE_CHECKING, List, Optional, Type
 
 import pytest
 
 from j5.backends import Backend, Environment
 from j5.boards.board import Board, BoardGroup
 
+if TYPE_CHECKING:  # pragma: nocover
```
Comment (trickeydan) [inline tests/boards/test_board.py]: Not sure if we need `# pragma: nocover` is needed here.

File: tests/boards/sr/v4/test_power_board.py [hunk 7]
```diff
@@ -18,6 +18,10 @@
 )
 from j5.components.piezo import Pitch
 
+if TYPE_CHECKING:  # pragma: nocover
```
Comment (trickeydan) [inline tests/boards/sr/v4/test_power_board.py]: Not sure if we need `# pragma: nocover` is needed here.

File: tests/backends/test_backend.py [hunk 8]
```diff
@@ -1,12 +1,15 @@
...
 
-from typing import List, Optional
+from typing import TYPE_CHECKING, List, Optional, Type
 
 import pytest
 
 from j5.backends import Backend, Environment
 from j5.boards import Board
 
+if TYPE_CHECKING:  # pragma: nocover
```
Comment (trickeydan) [inline tests/backends/test_backend.py]: Not sure if we need `# pragma: nocover` is needed here.

**Comment by kierdavis** [comment]
@trickeydan thanks for the feedback. I'd just copied the `pragma: nocover` directive from elsewhere in the codebase without really thinking about what it was used for.

**Pull request merged**